ssBind/posepicker/PosePicker.py

```diff
@@ -75,81 +73,65 @@ def pick_poses(
 
         scoredata = pd.read_csv(csv_scores, delimiter=",", header=0)
         PCA_Scores = pd.merge(df, scoredata, on="Index")
-        PCA_Scores.to_csv("PCA_Scores.csv", index=False, header=True)
 
+        label_data = []
         pcs = ["PC1", "PC2", "PC3"]
         for i in range(len(pcs)):
             for j in range(i + 1, len(pcs)):
-
+    
                 index_data = []
-
-                r(
-                    f"""
-                library(ggplot2)
-
-                df <- read.table('PCA_Scores.csv', header=TRUE, sep=",")
-
-                p <- ggplot(df, aes(x={pcs[i]}, y={pcs[j]}, z=Score)) +
-                    stat_summary_2d(fun=mean, binwidth = {self._binsize}) 
-
-                plot_data <- ggplot_build(p)
-                bin_data <- plot_data$data[[1]]
-                write.csv(bin_data, file = '{pcs[i]}_{pcs[j]}.csv', row.names = FALSE)
-                """
-                )
-
-                data = pd.read_csv(f"{pcs[i]}_{pcs[j]}.csv", delimiter=",", header=0)
-                raw_data = pd.read_csv("PCA_Scores.csv", delimiter=",", header=0)
-
-                df_sorted = data.sort_values(by="value")
-                top_bins = df_sorted.head(self._numbin)
-                extracted_data = top_bins[["xmin", "xmax", "ymin", "ymax"]]
-
+    
+                df = PCA_Scores
+    
+                # functions to find bin boundaries of each point based on binsize
+                xmin = lambda x: x - (x % self._binsize)
+                xmax =lambda x: x - (x % self._binsize) + self._binsize
+    
+                # set bin boundaries for each point in df (6 values, 2 for each PC)
+                df[f"{pcs[i]}_xmin"] = df[f"{pcs[i]}"].apply(xmin)
+                df[f"{pcs[i]}_xmax"] = df[f"{pcs[i]}"].apply(xmax)
+    
+                df[f"{pcs[j]}_ymin"] = df[f"{pcs[j]}"].apply(xmin)
+                df[f"{pcs[j]}_ymax"] = df[f"{pcs[j]}"].apply(xmax)
+    
```

Comment:
You can end the loop here - what is below only needs to be executed for PC1-PC2. And you can remove index_data or label_data and just keep one of them.